src/workflows/SnapshotActionMenu.tsx

```diff
@@ -0,0 +1,54 @@
+import { Clickable, Icon } from '@terra-ui-packages/components';
+import React, { ReactNode } from 'react';
+import { MenuButton } from 'src/components/MenuButton';
+import { makeMenuIcon, MenuTrigger } from 'src/components/PopupTrigger';
+
+export interface SnapshotActionMenuProps {
+  /**
+   * Whether the user is an owner of the workflow snapshot the actions in the
+   * menu are for. Controls whether the following operations are enabled: delete
+   */
+  isSnapshotOwner: boolean;
+
+  /** The action to be performed if the "Delete snapshot" button is pressed. */
+  onDelete: () => void;
+}
+
+/**
+ * A kebab (vertical three-dot) menu that displays buttons to perform actions on
+ * a workflow snapshot.
+ *
+ * Currently supported actions: delete
+ */
+const SnapshotActionMenu = (props: SnapshotActionMenuProps): ReactNode => {
+  const { isSnapshotOwner, onDelete } = props;
+
+  const notSnapshotOwnerTooltip = 'You must be an owner of this snapshot';
+
+  const menuContent = (
+    <MenuButton
+      disabled={!isSnapshotOwner}
```

Comment:
Out of curiosity, is there a reason why you decided to disable the delete option as opposed to hiding it all together?

Reply:
I did this for consistency with other kebab menus in Terra UI, such as for workspace actions:

<img width="545" alt="disabled delete workspace action image" src="https://github.com/user-attachments/assets/d6e5a2f5-3a83-4a6a-ab79-fe64ed48d3a1">